### DISCUSSION TITLE
[Core Protocol] Looking to have Decentralised Key Generation on Nodes to enable public key cryptography.

### DISCUSSION THREAD

**Discussion by vaibhavgeek** [ethereum-optimism/developers#527] [Core Protocol]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [ ] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I am application developer for [Kleo Network](https://www.kleo.network/), am trying to create an infrastructure where nodes on my app chain can create DGK decentralised generate key to create ED5519 signature scheme for any message / content. This DGK should happen using Shamir's secret sharing algorithm so that no single node has access to entire user data. 

 I also want them to run very specific docker container inside Intel SGX machines (TEEs). 

I am researching on how to get this done on OP stack, looking forward to resources and pointers that can help me along the way.

**Comment by opfocus** [comment]
cc @sbvegan

**Comment by Chomtana** [comment]
- OP Stack mainly uses ECDSA not ED5519
- The closest to this is the raft consensus on https://github.com/ethereum-optimism/optimism/tree/develop/op-conductor
- I haven't seen OP Stack or any EVM chain achieved threshold key
